### PR TITLE
Fix several NTFS decompression bugs

### PR DESCRIPTION
**Pull request by uckelman-sf**
This patch fixes decompression errors in NTFS compressed files that have an initialized size which differs from the logical size, and also allows decompression of an NTFS compressed file to continue in the event that a corrupted compression unit is encountered.

### CODE REVIEW

**Comment by bcarrier** [comment]
I had to revert this because it caused a regression in one of our test images.  The file ended up getting a few chunks of 0s were it used to have valid data.  The file was 100% initialized, so no special handling was required.

File: tsk/fs/ntfs.c [hunk 1]
```diff
@@ -1360,18 +1367,29 @@ ntfs_attr_walk_special(const TSK_FS_ATTR * fs_attr,
...
+                                    flags & TSK_FS_META_FLAG_ALLOC) ?
+                                "Allocated" : "Deleted");
+                            free(comp_unit);
+                            ntfs_uncompress_done(&comp);
+                            return 1;
+                        }
+
+                        // handle the initialized size
+                        int64_t remanining_init_size = fs_attr->nrd.initsize - off;
+                        if (remanining_init_size < comp.buf_size_b) {
```
Comment (bcarrier) [inline tsk/fs/ntfs.c]: I don't understand this check.  Is your goal to provide zeros for everything past the initialized area?

Comment (kastonework) [reply tsk/fs/ntfs.c]: Yes - the intention is to zero-fill after the init size is reached, to match Windows' behavior for the file.

**Comment by kastonework** [comment]
@bcarrier Is there any way to get a hold of the test image that's displaying the regression?

**Comment by uckelman-sf** [comment]
@bcarrier Is the test image where you spotted the problem one you could give us? @kastonework is investigating the problem, but it would help her if she had that.

**Comment by uckelman-sf** [comment]
PR #1276 addresses the bug causing the regression.

**Pull request merged**